Build failing on ubuntu 22.04

I assume I must be using clang and `-Werror` is set, but probably should just initialize the `tmp` vars.

```
matt@deepdrive:~$ uname -a
Linux deepdrive 5.10.120-tegra #1 SMP PREEMPT Wed Aug 9 15:29:32 CST 2023 aarch64 aarch64 aarch64 GNU/Linux

matt@deepdrive:~/src/msp$ cmake -B build -GNinja -DCMAKE_BUILD_TYPE=Release
-- Configuring done
-- Generating done
-- Build files have been written to: /home/matt/src/msp/build

matt@deepdrive:~/src/msp$ cmake --build build
[1/10] Building CXX object CMakeFiles/client_read_test.dir/examples/client_read_test.cpp.o
FAILED: CMakeFiles/client_read_test.dir/examples/client_read_test.cpp.o 
/usr/bin/c++  -DASIO_HAS_CSTDINT -DASIO_HAS_STD_ADDRESSOF -DASIO_HAS_STD_ARRAY -DASIO_HAS_STD_SHARED_PTR -DASIO_HAS_STD_TYPE_TRAITS -DASIO_STANDALONE -I../inc/msp -O3 -DNDEBUG   -Wall -Wextra -Wpedantic -Werror -std=c++17 -MD -MT CMakeFiles/client_read_test.dir/examples/client_read_test.cpp.o -MF CMakeFiles/client_read_test.dir/examples/client_read_test.cpp.o.d -o CMakeFiles/client_read_test.dir/examples/client_read_test.cpp.o -c ../examples/client_read_test.cpp
In file included from ../inc/msp/Client.hpp:11,
                 from ../examples/client_read_test.cpp:1:
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::DebugMessage::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  275 |         int8_t tmp;
      |                ^~~
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::PidNames::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::BoxNames::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
cc1plus: all warnings being treated as errors
[2/10] Building CXX object CMakeFiles/msp_fcu.dir/src/FlightController.cpp.o
FAILED: CMakeFiles/msp_fcu.dir/src/FlightController.cpp.o 
/usr/bin/c++  -DASIO_HAS_CSTDINT -DASIO_HAS_STD_ADDRESSOF -DASIO_HAS_STD_ARRAY -DASIO_HAS_STD_SHARED_PTR -DASIO_HAS_STD_TYPE_TRAITS -DASIO_STANDALONE -Dmsp_fcu_EXPORTS -I../inc/msp -O3 -DNDEBUG -fPIC   -Wall -Wextra -Wpedantic -Werror -std=c++17 -MD -MT CMakeFiles/msp_fcu.dir/src/FlightController.cpp.o -MF CMakeFiles/msp_fcu.dir/src/FlightController.cpp.o.d -o CMakeFiles/msp_fcu.dir/src/FlightController.cpp.o -c ../src/FlightController.cpp
In file included from ../inc/msp/Client.hpp:11,
                 from ../inc/msp/FlightController.hpp:4,
                 from ../src/FlightController.cpp:1:
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::BuildInfo::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  275 |         int8_t tmp;
      |                ^~~
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::FcVariant::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::BoardInfo::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::BoxNames::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
cc1plus: all warnings being treated as errors
[3/10] Building CXX object CMakeFiles/client_async_test.dir/examples/client_async_test.cpp.o
FAILED: CMakeFiles/client_async_test.dir/examples/client_async_test.cpp.o 
/usr/bin/c++  -DASIO_HAS_CSTDINT -DASIO_HAS_STD_ADDRESSOF -DASIO_HAS_STD_ARRAY -DASIO_HAS_STD_SHARED_PTR -DASIO_HAS_STD_TYPE_TRAITS -DASIO_STANDALONE -I../inc/msp -O3 -DNDEBUG   -Wall -Wextra -Wpedantic -Werror -std=c++17 -MD -MT CMakeFiles/client_async_test.dir/examples/client_async_test.cpp.o -MF CMakeFiles/client_async_test.dir/examples/client_async_test.cpp.o.d -o CMakeFiles/client_async_test.dir/examples/client_async_test.cpp.o -c ../examples/client_async_test.cpp
In file included from ../inc/msp/Client.hpp:11,
                 from ../examples/client_async_test.cpp:1:
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::DebugMessage::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  275 |         int8_t tmp;
      |                ^~~
../inc/msp/ByteVector.hpp: In member function ‘void msp::client::Subscription<T>::decode(msp::ByteVector&) const [with T = msp::msg::DebugMessage]’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::BoxNames::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::PidNames::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
cc1plus: all warnings being treated as errors
[4/10] Building CXX object CMakeFiles/fcu_test.dir/examples/fcu_test.cpp.o
FAILED: CMakeFiles/fcu_test.dir/examples/fcu_test.cpp.o 
/usr/bin/c++  -DASIO_HAS_CSTDINT -DASIO_HAS_STD_ADDRESSOF -DASIO_HAS_STD_ARRAY -DASIO_HAS_STD_SHARED_PTR -DASIO_HAS_STD_TYPE_TRAITS -DASIO_STANDALONE -I../inc/msp -O3 -DNDEBUG   -Wall -Wextra -Wpedantic -Werror -std=c++17 -MD -MT CMakeFiles/fcu_test.dir/examples/fcu_test.cpp.o -MF CMakeFiles/fcu_test.dir/examples/fcu_test.cpp.o.d -o CMakeFiles/fcu_test.dir/examples/fcu_test.cpp.o -c ../examples/fcu_test.cpp
In file included from ../inc/msp/Client.hpp:11,
                 from ../inc/msp/FlightController.hpp:4,
                 from ../examples/fcu_test.cpp:1:
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::DebugMessage::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  275 |         int8_t tmp;
      |                ^~~
../inc/msp/ByteVector.hpp: In member function ‘void msp::client::Subscription<T>::decode(msp::ByteVector&) const [with T = msp::msg::DebugMessage]’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::BoxNames::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
../inc/msp/ByteVector.hpp: In member function ‘virtual bool msp::msg::PidNames::decode(const msp::ByteVector&)’:
../inc/msp/ByteVector.hpp:275:16: error: ‘tmp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.```

--- a/.github/workflows/cmake.yml
+++ b/.github/workflows/cmake.yml
@@ -5,6 +5,7 @@ on:
     branches: [ master ]
   pull_request:
     branches: [ master ]
+  workflow_dispatch:
 
 env:
   BUILD_TYPE: Release

--- a/.github/workflows/cmake.yml
+++ b/.github/workflows/cmake.yml
@@ -18,14 +18,14 @@ jobs:
       fail-fast: false
       matrix:
         config:
-          - { name: "Linux", os: ubuntu-latest, test: ON }
-          - { name: "Windows", os: windows-latest, test: OFF }
-          - { name: "macOS", os: macos-latest, test: OFF }
+          - { name: "Linux", os: ubuntu-24.04, test: ON }
+          - { name: "Windows", os: windows-2025, test: OFF }
+          - { name: "macOS", os: macos-15, test: OFF }
 
     steps:
     - name: install Linux dependencies
       if: runner.os == 'Linux'
-      run: sudo apt install -y --no-install-recommends ninja-build libasio-dev clang-format-12 libgtest-dev
+      run: sudo apt install -y --no-install-recommends ninja-build libasio-dev clang-format-18 libgtest-dev
 
     - name: install Microsoft Visual C++ (Windows)
       uses: ilammy/msvc-dev-cmd@v1

--- a/.github/workflows/cmake.yml
+++ b/.github/workflows/cmake.yml
@@ -44,7 +44,7 @@ jobs:
       if: runner.os == 'macOS'
       run: brew install ninja asio
 
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
 
     - name: Clang-Format Style Check
       if: runner.os == 'Linux'

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.16)
 project(msp VERSION 3.4.0 LANGUAGES CXX)
 
 set(PROJECT_DESCRIPTION "Implementation of the MultiWii Serial Protocol (MSP) for MultiWii and Cleanflight flight controller")

--- a/format_source_files.sh
+++ b/format_source_files.sh
@@ -1,2 +1,2 @@
 #!/bin/sh
-find . -regex '.*\.\(cpp\|hpp\)' -exec clang-format-12 -style=file -i {} \;
+find . -regex '.*\.\(cpp\|hpp\)' -exec clang-format-18 -style=file -i {} \;

--- a/inc/msp/ByteVector.hpp
+++ b/inc/msp/ByteVector.hpp
@@ -272,7 +272,7 @@ public:
         if(count > unpacking_remaining()) return false;
         bool rc = true;
         val.clear();
-        int8_t tmp;
+        int8_t tmp = {};
         for(size_t i = 0; i < count; ++i) {
             rc &= unpack(tmp);
             val += tmp;

--- a/inc/msp/Value.hpp
+++ b/inc/msp/Value.hpp
@@ -1,6 +1,7 @@
 #ifndef VALUE_HPP
 #define VALUE_HPP
 
+#include <cstdint>
 #include <iostream>
 #include <utility>
 

--- a/inc/msp/msp_msg.hpp
+++ b/inc/msp/msp_msg.hpp
@@ -3320,7 +3320,7 @@ struct ActiveBoxes : public Message {
                     if(box_conf() & (1 << (iaux * 3 + ip)))
                         aux_sp[iaux].insert(SwitchPosition(ip));
                 }  // each position (L,M,H)
-            }      // each aux switch
+            }  // each aux switch
             box_pattern.push_back(aux_sp);
         }  // each box
         return rc;

